### ISSUE TITLE
"docker-compose up -d" fails in ipv6nat setup

### ISSUE THREAD

**Issue by tchwpkgorg** [mailcow/mailcow-dockerized#316]
I'm trying to run mailcow-dockerized in a LXD container. Unfortunately, it fails with:

```
root@mail:~/mailcow-dockerized# docker-compose up -d
Creating network "mailcowdockerized_mailcow-network" with driver "bridge"
Creating volume "mailcowdockerized_crypt-vol-1" with default driver
Creating volume "mailcowdockerized_redis-vol-1" with default driver
Creating volume "mailcowdockerized_vmail-vol-1" with default driver
Creating volume "mailcowdockerized_postfix-vol-1" with default driver
Creating volume "mailcowdockerized_rspamd-vol-1" with default driver
Creating volume "mailcowdockerized_mysql-vol-1" with default driver
Creating volume "mailcowdockerized_dkim-vol-1" with default driver
Creating mailcowdockerized_mysql-mailcow_1 ... 
Creating mailcowdockerized_ipv6nat_1 ... 
Creating mailcowdockerized_clamd-mailcow_1 ... 
Creating mailcowdockerized_mysql-mailcow_1
Creating mailcowdockerized_clamd-mailcow_1
Creating mailcowdockerized_ipv6nat_1 ... error

Creating mailcowdockerized_clamd-mailcow_1 ... done
Creating mailcowdockerized_bind9-mailcow_1 ... 
Creating mailcowdockerized_bind9-mailcow_1 ... done
Creating mailcowdockerized_sogo-mailcow_1 ... 
Creating mailcowdockerized_redis-mailcow_1 ... 
Creating mailcowdockerized_rmilter-mailcow_1 ... 
Creating mailcowdockerized_dovecot-mailcow_1 ... 
Creating mailcowdockerized_memcached-mailcow_1 ... 
Creating mailcowdockerized_postfix-mailcow_1 ... 
Creating mailcowdockerized_rmilter-mailcow_1
Creating mailcowdockerized_redis-mailcow_1
Creating mailcowdockerized_sogo-mailcow_1
Creating mailcowdockerized_memcached-mailcow_1
Creating mailcowdockerized_dovecot-mailcow_1
Creating mailcowdockerized_memcached-mailcow_1 ... done
Creating mailcowdockerized_php-fpm-mailcow_1 ... 
Creating mailcowdockerized_php-fpm-mailcow_1 ... done
Creating mailcowdockerized_nginx-mailcow_1 ... 
Creating mailcowdockerized_nginx-mailcow_1 ... done
Creating mailcowdockerized_rspamd-mailcow_1 ... 
Creating mailcowdockerized_rspamd-mailcow_1 ... done

ERROR: for ipv6nat  Cannot start service ipv6nat: linux runtime spec devices: lstat /dev/.lxc/proc/416/fdinfo/23: no such file or directory
ERROR: Encountered errors while bringing up the project.
```

The "standalone" mailcow version didn't have this issue.

**Comment by andryyy** [comment]
Seems to be fixed in Docker 17.06. But it is not yet released.

**Comment by mkuron** [comment]
This issue: https://github.com/moby/moby/issues/32968.

For the time being, until 17.06.0 is released, you can remove/comment the ipv6nat container in docker-compose.yml. If your server has IPv6, you might want to disable it as without the container, all incoming IPv6 connections appear to be coming from your Docker host, thus defeating many security measures.

**Comment by tchwpkgorg** [comment]
Using IPv6 with all connections being seen as coming from the localhost / local server defeats the purpose of using IPv6, heh.

Is there any fix for that in mailcow? Seems like architecture flaw of mailcow-dockerized O_o

**Comment by andryyy** [comment]
All connections being seen as local? Where?

**Comment by tchwpkgorg** [comment]
I was referring to this: 

```If your server has IPv6, you might want to disable it as without the container, all incoming IPv6 connections appear to be coming from your Docker host, thus defeating many security measures.```

**Comment by mkuron** [comment]
> Using IPv6 with all connections being seen as coming from the localhost / local server defeats the purpose of using IPv6, heh.
> Is there any fix for that in mailcow? Seems like architecture flaw of mailcow-dockerized O_o

It's a flaw in Docker, not in Mailcow. But we added the ipv6nat container a few weeks ago, which makes Docker's expose mechanism work with IPv6 and you should not see IPv6 connections as originating from the host anymore.

> I was referring to this:

This issue is only about using it with Docker running inside LXD instead of directly on the Linux kernel, which is not exactly a common/standard configuration. Evidently that breaks the ipv6nat container in the current Docker version but will be fixed in Docker 17.06.0.

**Comment by tchwpkgorg** [comment]
Thanks for explanation.

I think we can close the ticket for now, as it will eventually work when newer docker is released.
